File: arrays/max_water/max_water.py
""" Solution Module """


class Solution:
    """Solution Class"""

    def run_solution(self, test_input):
        """Solution Method"""
        max_vol = 0

        left = 0
        right = len(test_input) - 1

        def check_and_update(left, right):
            curr_max = min(test_input[left], test_input[right]) * (right - left)
            return max(curr_max, max_vol)

        while left < right:
            max_vol = check_and_update(left, right)
            if test_input[left] <= test_input[right]:
                left += 1
            else:
                right -= 1

        return max_vol

File: arrays/max_water/max_water.ts
import * as test from 'node:test';

export class Solution {
    /**
     * @param {string} testInput
     * @return {any} Returns either boolean to validate the test cases or a custom result.
     */
    run_solution(testInput: any) {
        let maxVol: number = 0;

        let left = 0;
        let right = test.length;

        while (left < right) {
            const currMax =
                Math.min(testInput[left], testInput[right]) * (right - left);
            maxVol = Math.max(currMax, maxVol);
            if (testInput[left] < testInput[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxVol;
    }
}
